server/src/test/java/org/elasticsearch/cluster/coordination/ClusterBootstrapServiceTests.java

```diff
@@ -648,9 +654,81 @@ public void testFailBootstrapNonMasterEligibleNodeWithSingleNodeDiscovery() {
         assertThat(
             expectThrows(
                 IllegalArgumentException.class,
-                () -> new ClusterBootstrapService(settings.build(), transportService, () -> emptyList(), () -> false, vc -> fail())
+                () -> new ClusterBootstrapService(settings.build(), transportService, Collections::emptyList, () -> false, vc -> fail())
             ).getMessage(),
             containsString("node with [discovery.type] set to [single-node] must be master-eligible")
         );
     }
+
+    public void testBootstrapStateLogging() throws IllegalAccessException {
+        final var mockAppender = new MockLogAppender();
+        mockAppender.start();
+        final var serviceLogger = LogManager.getLogger(ClusterBootstrapService.class);
+        Loggers.addAppender(serviceLogger, mockAppender);
+
+        try {
+            mockAppender.addExpectation(
+                new MockLogAppender.SeenEventExpectation(
+                    "fresh node message",
+                    ClusterBootstrapService.class.getCanonicalName(),
+                    Level.INFO,
+                    "this node has not joined a bootstrapped cluster yet; [cluster.initial_master_nodes] is set to [node1, node2]"
+                )
+            );
+
+            new ClusterBootstrapService(
+                Settings.builder().putList(INITIAL_MASTER_NODES_SETTING.getKey(), "node1", "node2").build(),
+                transportService,
+                Collections::emptyList,
+                () -> false,
+                vc -> { throw new AssertionError("should not be called"); }
+            ).logBootstrapState(Metadata.builder().clusterUUIDCommitted(false).build());
```

Comment:
Can we randomly add a `clusterUUID` too?

Reply:
Indeed we can, see [9e17662](https://github.com/elastic/elasticsearch/pull/86101/commits/9e1766286493e4979f4d33f3396da1bd9ab838b4).